content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-12.md

```diff
@@ -0,0 +1,299 @@
+---
+reviewers:
+- sig-cluster-lifecycle
+title: Upgrading kubeadm clusters from v1.11 to v1.12
+content_template: templates/task
+---
+
+{{% capture overview %}}
+
+This page explains how to upgrade a Kubernetes cluster created with `kubeadm` from version 1.11.x to version 1.12.x, and from version 1.12.x to 1.12.y, where `y > x`.
+
+{{% /capture %}}
+
+{{% capture prerequisites %}}
+
+- You need to have a `kubeadm` Kubernetes cluster running version 1.11.0 or later. 
+  [Swap must be disabled][swap]. 
+  The cluster should use a static control plane and etcd pods.
+- Make sure you read the [release notes](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.12.md) carefully.
+- Make sure to back up any important components, such as app-level state stored in a database. 
+  `kubeadm upgrade` does not touch your workloads, only components internal to Kubernetes, but backups are always a best practice.
+
+
+[swap]: https://serverfault.com/questions/684771/best-way-to-disable-swap-in-linux
+### Additional information
+
+- All containers are restarted after upgrade, because the container spec hash value is changed.
+- You can upgrade only from one minor version to the next minor version. 
+  That is, you cannot skip versions when you upgrade. 
+  For example, you can upgrade only from 1.10 to 1.11, not from 1.9 to 1.11.
+- The default DNS provider in version 1.12 is [CoreDNS](https://coredns.io/) rather than [kube-dns](https://github.com/kubernetes/dns).
+  To keep `kube-dns`, pass `--feature-gates=CoreDNS=false` to `kubeadm upgrade apply`.
+
+{{% /capture %}}
+
+{{% capture steps %}}
+
+## Upgrade the control plane
+
+1.  On your master node, upgrade kubeadm:
+
+    {{< tabs name="k8s_install" >}}
+    {{% tab name="Ubuntu, Debian or HypriotOS" %}}
+    apt-get update
+    apt-get upgrade -y kubelet kubeadm
+    {{% /tab %}}
+    {{% tab name="CentOS, RHEL or Fedora" %}}
+    yum upgrade -y kubeadm --disableexcludes=kubernetes
+    {{% /tab %}}
+    {{< /tabs >}}
+    
+1.  Verify that the download works and has the expected version:
+
+    ```shell
+    kubeadm version
+    ```
+
+1.  On the master node, run:
+
+    ```shell
+    kubeadm upgrade plan
+    ```
+
+    You should see output similar to this:
+
+    ```shell
+    [preflight] Running pre-flight checks.
+    [upgrade] Making sure the cluster is healthy:
+    [upgrade/config] Making sure the configuration is correct:
+    [upgrade/config] Reading configuration from the cluster...
+    [upgrade/config] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -oyaml'
+    [upgrade] Fetching available versions to upgrade to
+    [upgrade/versions] Cluster version: v1.11.3
+    [upgrade/versions] kubeadm version: v1.12.0-rc.1
```

Comment:
could you just nix all the `-rc.1` bits, b/c the users will never see that on releases.